diff --git a/wrappers/generateWrappers.py b/wrappers/generateWrappers.py
index cd22e19de..8742dcbfb 100644
--- a/wrappers/generateWrappers.py
+++ b/wrappers/generateWrappers.py
@@ -235,7 +235,7 @@ class CHeaderGenerator(WrapperGenerator):
         shortClassName = stripOpenMMPrefix(className)
         typeName = convertOpenMMPrefix(className)
         destructorName = '~'+shortClassName
-        isAbstract = any('virt' in method.attrib and method.attrib['virt'] == 'pure-virtual' for method in classNode.getiterator('memberdef'))
+        isAbstract = any('virt' in method.attrib and method.attrib['virt'] == 'pure-virtual' for method in classNode.iter('memberdef'))

         if not isAbstract:
             # Write constructors
@@ -496,7 +496,7 @@ class CSourceGenerator(WrapperGenerator):
         shortClassName = stripOpenMMPrefix(className)
         typeName = convertOpenMMPrefix(className)
         destructorName = '~'+shortClassName
-        isAbstract = any('virt' in method.attrib and method.attrib['virt'] == 'pure-virtual' for method in classNode.getiterator('memberdef'))
+        isAbstract = any('virt' in method.attrib and method.attrib['virt'] == 'pure-virtual' for method in classNode.iter('memberdef'))

         if not isAbstract:
             # Write constructors
@@ -959,7 +959,7 @@ class FortranHeaderGenerator(WrapperGenerator):
         shortClassName = stripOpenMMPrefix(className)
         typeName = convertOpenMMPrefix(className)
         destructorName = '~'+shortClassName
-        isAbstract = any('virt' in method.attrib and method.attrib['virt'] == 'pure-virtual' for method in classNode.getiterator('memberdef'))
+        isAbstract = any('virt' in method.attrib and method.attrib['virt'] == 'pure-virtual' for method in classNode.iter('memberdef'))

         if not isAbstract:
             # Write constructors
@@ -1529,7 +1529,7 @@ class FortranSourceGenerator(WrapperGenerator):
         shortClassName = stripOpenMMPrefix(className)
         typeName = convertOpenMMPrefix(className)
         destructorName = '~'+shortClassName
-        isAbstract = any('virt' in method.attrib and method.attrib['virt'] == 'pure-virtual' for method in classNode.getiterator('memberdef'))
+        isAbstract = any('virt' in method.attrib and method.attrib['virt'] == 'pure-virtual' for method in classNode.iter('memberdef'))

         if not isAbstract:
             # Write constructors
