--- a/platforms/reference/src/ReferenceTabulatedFunction.cpp
+++ b/platforms/reference/src/ReferenceTabulatedFunction.cpp
@@ -36,6 +36,7 @@
 #ifdef _MSC_VER
 
 #if _MSC_VER < 1800
+#include <minmax.h>^M
 /**
  * We need to define this ourselves, since Visual Studio is missing round() from cmath.
  */
@@ -43,6 +44,7 @@ static int round(double x) {
     return (int) (x+0.5);
 }
 #else
+#include <minmax.h>^M
 #include <cmath>
 #endif  // MSC_VER < 1800
